### DISCUSSION TITLE
ytmusicapi enables easy bulk deletion of unwanted tracks

### DISCUSSION THREAD

**Discussion by mavaddat** [sigma67/ytmusicapi#424] [Show and tell]
I uploaded two-hundred songs without the proper ID3v2 data set on the tracks. I tried to write a JavaScript console procedure to delete the accidentally added tracks, but I couldn't easily whip one up that would run successfully in the browser. So, I Googled for the YouTube Music API and happily stumbled upon this great project.

In an interactive prompt, I ran the following commands:

```python3
from ytmusicapi import YTMusic
from ytmusicapi import setup_oauth

auth = setup_oauth()
ytmusic = YTMusic(auth)
songsToDelete = ytmusic.get_library_upload_songs(225)  # Not quite sure which is the last track, so I grabbed more than necessary

songsToDelete[204]  # Verify the last song I want to delete

for i in range(205):
    ytmusic.delete_upload_entity(songsToDelete[i]["entityId"])
```

Which results in:

```
'STATUS_SUCCEEDED'
```

I checked my YouTube Music and verified that it removed all (and only) the songs I wanted to remove. So cool!

**Comment by apastel** [comment]
I made a Windows GUI app using this API that does these bulk deletions, since Google stupidly did not provide a bulk deletion mechanism for YT Music.

https://github.com/apastel/ytmusic-deleter

**Reply by mavaddat** [reply]
![Standing Ovation Applause GIF By The Maury Show](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExY2htbGNjY25hcGl6d2U0ZDVweWIwdXNvMGY5dzlwaW9zZmtiMGlzeCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/l3V0wkQ2KKcAeW8Cs/giphy.gif)

**Reply by apastel** [reply]
Lol. It still uses the browser authentication but I'm in the process of switching it over to the OAuth flow.
